Week7/CatApi/app/src/main/java/com/example/catapi/ui/SecondFragment.kt

```diff
@@ -0,0 +1,195 @@
+                        arrayOf("Nothing to show")
+                    )
+                }
+                breedAdapter?.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+                sp_breed.adapter = breedAdapter
+            }
+
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        model = activity?.let { ViewModelProvider(it).get(MyViewModel::class.java) }!!
+    }
+
+    override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_second, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        // ---------------------------------------------------------------------------Show options in the spinner genre
+        val genreAdapter = activity?.let {
+            ArrayAdapter(
+                it,
+                android.R.layout.simple_spinner_dropdown_item,
+                genreList
+            )
+        }
+        genreAdapter?.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        sp_genre.adapter = genreAdapter
+        obtainBreedList()
+        //-------------------------------------------------------------------------------------------------------------
+
+        //----------------------------------- When the image view is tapped, new cat of the breed selected is generated
+        iv_cat_random.setOnClickListener {
+            searchCat(sp_breed.selectedItem.toString())
+        }
+        //-------------------------------------------------------------------------------------------------------------
+        //------------------------------------------------------------------------------------ Button for add a new cat
+        bt_new_cat.setOnClickListener {
+            if (!emptyName()){
+                val cat = Cat(et_cat_name.text.toString(),sp_breed.selectedItem.toString(),sp_genre.selectedItem.toString(),image)
+                // Add changes in the database
+                model.addCat(cat)
+
+                // Show a message
+                Toast.makeText(activity, "A new item has been added!", Toast.LENGTH_SHORT).show()
+                view.findNavController().navigate(R.id.action_SecondFragment_self)
+            }else
+                Toast.makeText(activity,"The cat must have a name!",Toast.LENGTH_SHORT).show()
+        }
+        //-------------------------------------------------------------------------------------------------------------
+    }
+
+    private fun emptyName() : Boolean{
+        return et_cat_name.text.isEmpty()
+    }
+
+
+    private fun searchCat(breedSelected : String){
+        GlobalScope.launch(Dispatchers.Main) {
```

Comment:
Once again, avoid global scope if possible, perhaps use the lifecycle or main scope?